lib/ansible/parsing/vault/__init__.py

```diff
@@ -658,7 +658,10 @@ def decrypt_and_get_vault_id(self, vaulttext, filename=None, obj=None):
         b_vaulttext = to_bytes(vaulttext, errors='strict', encoding='utf-8')
 
         if self.secrets is None:
-            raise AnsibleVaultError("A vault password must be specified to decrypt data")
+            msg = "A vault password must be specified to decrypt data"
+            if filename:
+                msg += "in file %s" % to_native(filename)
```

Comment:
```suggestion
                msg += " in file %s" % to_native(filename)
```

Reply:
Missing space added in 51deb16481858a26479d1f983eb51888b79714b7